lib/Reader/readerir.cpp

```diff
@@ -575,6 +576,92 @@ void GenIR::callMonitorHelper(bool IsEnter) {
                  MethodSyncHandle, (IRNode *)SyncFlag);
 }
 
+void GenIR::insertIRForUnmanagedCallFrame() {
+  const struct CORINFO_EE_INFO::InlinedCallFrameInfo &CallFrameInfo =
+      JitContext->EEInfo.inlinedCallFrameInfo;
+  LLVMContext &LLVMContext = *JitContext->LLVMContext;
+  Type *Int8Ty = Type::getInt8Ty(LLVMContext);
+  Type *Int32Ty = Type::getInt32Ty(LLVMContext);
+  Type *Int8PtrTy = getUnmanagedPointerType(Int8Ty);
+  IRBuilder<>::InsertPoint SavedInsertPoint = LLVMBuilder->saveIP();
+
+  // Mark this function as requiring a frame pointer and as using GC.
+  Function->addFnAttr("no-frame-pointer-elim-non-leaf");
```

Comment:
Setting the target options offers module-level control; using the attribute offers function-level control. I would lean towards the attribute, assuming that it also does the correct thing in the case that we encounter leaf functions which require a frame pointer.